### PR TITLE
feat(experiments): add variant screenshots

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes
Users can now upload screenshots to visualize each variant in the Experiment UI.

https://github.com/user-attachments/assets/c7e7cc6f-c855-4a70-91b2-0e2779e1fa5d

## How did you test this code?
👀

### CODE REVIEW

File: frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                    multiple={false}
+                    onChange={setFilesToUpload}
+                    loading={uploading}
+                    value={filesToUpload}
+                    callToAction={
+                        <div className="relative">
+                            <div className="Lettermark Lettermark--xlarge Lettermark--rounded Lettermark--variant-4">
+                                <IconUpload />
+                            </div>
+                        </div>
```
Comment (Twixes) [inline frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx]: Plugging into the internals of another component is fragile – changes to Lettermark styling in the future are likely to break `VariantPreviewImage`. On a different note, there is very poor contrast between the upload icon and its background.

I propose a more conventional file input, which also lets us be more descriptive about this preview feature:
```suggestion
                            <>
                                <IconUpload className="text-2xl" />
                                <span>Upload a preview of this variant's UI</span>
                            </>
```

Looking like this:
Screenshot 2024-10-07 at 12 05 34

File: frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                    }
+                />
+            ) : (
+                <div className="relative">
+                    <>
+                        <div className="text-muted inline-flex flow-row items-center gap-1 cursor-pointer">
+                            <div onClick={() => setIsModalOpen(true)} className="cursor-pointer relative">
```
Comment (Twixes) [inline frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx]: I had no idea that I must click on this tile to see the preview in a large scale – only realized this looking at the code! In other file upload contexts you click on the image to _change_ it (e.g. org logo or Hog destinations). But an easy way to create the affordance for image expansion is `cursor: zoom-in`:

Screenshot 2024-10-07 at 12 21 55

File: frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                                <div className="relative flex overflow-hidden select-none size-16 rounded before:absolute before:inset-0 before:border before:rounded">
```
Comment (Twixes) [inline frontend/src/scenes/experiments/ExperimentView/VariantPreviewImage.tsx]: Screenshot 2024-10-07 at 12 10 15

This is so so small – when clearly we're not hard-pressed for space, looking at all the whitespace – that it's hard to see the point of the preview.

**Comment by jurajmajerik** [comment]
Feedback addressed, thanks @Twixes!

**Review by Twixes — APPROVED** [review]
👌

**Pull request merged**